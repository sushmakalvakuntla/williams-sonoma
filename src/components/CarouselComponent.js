import React from 'react';
import { Carousel, Image } from 'react-bootstrap';

const CarouselComponent = ({ images }) => {
  return (
    <Carousel>
      {images?.map((image, index) => (
        <Carousel.Item key={index.toString()}>
          <Image src={image.href} />
        </Carousel.Item>
      ))}
    </Carousel>
  );
};

export default CarouselComponent;
