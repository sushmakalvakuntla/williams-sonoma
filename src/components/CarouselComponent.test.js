import React from 'react';
import { shallow } from 'enzyme';
import CarouselComponent from './CarouselComponent';

it('renders CustomCarousel correctly', () => {
  const images = [
    {
      size: "m",
      meta: "",
      alt: "",
      rel: "althero",
      width: 363,
      href: "https://assets.weimgs.com/weimgs/ab/images/wcm/products/202111/0061/walnut-led-desk-lamp-2-m.jpg",
      height: 363
    },
  ];

  const component = shallow(<CarouselComponent images={images}/>);
  expect(component).toMatchSnapshot();
});
