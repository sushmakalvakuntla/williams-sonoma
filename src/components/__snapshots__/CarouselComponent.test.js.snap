// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomCarousel correctly 1`] = `
<Carousel
  controls={true}
  defaultActiveIndex={0}
  fade={false}
  indicators={true}
  interval={5000}
  keyboard={true}
  nextIcon={
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
  }
  nextLabel="Next"
  pause="hover"
  prevIcon={
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
  }
  prevLabel="Previous"
  slide={true}
  touch={true}
  wrap={true}
>
  <CarouselItem
    key="0"
  >
    <Image
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="https://assets.weimgs.com/weimgs/ab/images/wcm/products/202111/0061/walnut-led-desk-lamp-2-m.jpg"
      thumbnail={false}
    />
  </CarouselItem>
</Carousel>
`;
